server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -211,6 +213,12 @@ public DocsStats docStats() {
         }
     }
 
+    public IgnoredFieldStats ignoredFieldStats() {
+        try (Searcher searcher = acquireSearcher("ignored_field", SearcherScope.INTERNAL)) {
```

Comment:
@martijnvg this is the logic I was talking about. I had to add a new `source` to avoid `FrozenStorageDeciderIT#testScale` fail. If I use "doc_stats" as source if fails at the assertion

```
assert false : "doc stats are eagerly loaded";
```

in `FrozenEngine#openSearcher`

Reply:
I think the right thing to do would be to disallow calling this on frozen tiers...I don't see a reason why a user might be interested in knowing ignored fields for data that is in frozen...am I wrong?